### DISCUSSION TITLE
How can I pass values into a spec during compile?

### DISCUSSION THREAD

**Discussion by laduke** [microsoft/typespec#2915] [Q&A]
I for example, want to pass a version into my service definition

```
@service({
  title: "My Service",
  version: {VersionFromSomewhere},
})
```

I can get the value into an env var, a cli flag, even a .js file, but I haven't been able to figure out how to import/ interpolate it into the tsp file.

**Comment by bterlson** [comment]
This is not a feature we have, but could be added, I've filed an issue. The best thing I can think of is generating a "variables.tsp" out of band that you import into your spec.

**Comment by laduke** [comment]
thanks!

simplest templating I could come up with without adding any new tooling:

```json
    "variables": "printf 'alias Version = \"%s\";\n' $npm_package_version > variables.tsp",
```

```sh
npm run variables
cat variables.tsp          
alias Version = "1.0.1";
```
